Unit tests failures

### What happened?

A few unit tests fail on my local machine.

### What are the steps to reproduce the bug?

```bash
git clone git@github.com:ecmwf/earthkit-data.git
cd earthkit-data
mamba env create -f environment.yml -n earthkit-data
mamba activate earthkit-data
pip install -e .
make unit-tests
```

### Version

0.1.dev317+g102741b

### Platform (OS and architecture)

Darwin MacBook-Pro-3.local 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:21:56 PDT 2023; root:xnu-8796.141.3~6/RELEASE_X86_64 x86_64

### Relevant log output

```shell
====================================================== short test summary info ======================================================
FAILED tests/core/test_metadata.py::test_grib_metadata_get - assert 196 == 192
FAILED tests/grib/test_grib_metadata.py::test_grib_metadata_namespace[file] - AssertionError: assert 190 == 186
FAILED tests/numpy_fs/test_numpy_fs_metadata.py::test_numpy_fs_metadata_keys - assert 187 == 183
FAILED tests/numpy_fs/test_numpy_fs_metadata.py::test_numpy_fs_metadata_namespace - AssertionError: assert 181 == 177
==================== 4 failed, 925 passed, 39 skipped, 33 deselected, 7 xfailed, 8 warnings in 74.20s (0:01:14) =====================
```


### Accompanying data

_No response_

### Organisation

B-Open / CADS-EQC

--- a/earthkit/data/readers/grib/metadata.py
+++ b/earthkit/data/readers/grib/metadata.py
@@ -365,6 +365,12 @@ class RestrictedGribMetadata(GribMetadata):
 
     EKD_NAMESPACE = "grib"
     INTERNAL_KEYS = [
+        "minimum",
+        "maximum",
+        "average",
+        "standardDeviation",
+        "skewness",
+        "kurtosis",
         "min",
         "max",
         "avg",

--- a/tests/core/test_metadata.py
+++ b/tests/core/test_metadata.py
@@ -122,19 +122,22 @@ def test_grib_metadata_get():
     ds = from_source("file", earthkit_examples_file("test.grib"))
     md = ds[0].metadata()
 
-    assert len(md) == 192
+    # the number/order of metadata keys can vary with the ecCodes version
+    md_num = len(md)
+    assert md_num > 100
+
     keys = list(md.keys())
-    assert len(keys) == 192
-    assert keys[0] == "globalDomain"
-    assert keys[10] == "wrongPadding"
+    assert len(keys) == md_num
+    assert "shortName" in keys
+    assert "maximum" in keys
 
     keys = [k for k in md]
-    assert len(keys) == 192
-    assert keys[0] == "globalDomain"
-    assert keys[10] == "wrongPadding"
+    assert len(keys) == md_num
+    assert "shortName" in keys
+    assert "maximum" in keys
 
     items = {k: v for k, v in md.items()}
-    assert len(items) == 192
+    assert len(items) == md_num
     assert items["shortName"] == "2t"
     assert items["typeOfLevel"] == "surface"
 

--- a/tests/grib/test_grib_metadata.py
+++ b/tests/grib/test_grib_metadata.py
@@ -421,27 +421,31 @@ def test_grib_metadata_namespace(mode):
     }
     assert r == ref
 
+    # The number/order of metadata keys can vary with the ecCodes version.
+    # The same is true for the namespaces.
+
     r = f[0].metadata(namespace=None)
     assert isinstance(r, dict)
-    assert len(r) == 186
+    md_num = len(r)
+    assert md_num > 100
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 
     r = f[0].metadata(namespace=[None])
     assert isinstance(r, dict)
-    assert len(r) == 186
+    assert len(r) == md_num
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 
     r = f[0].metadata(namespace="")
     assert isinstance(r, dict)
-    assert len(r) == 186
+    assert len(r) == md_num
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 
     r = f[0].metadata(namespace=[""])
     assert isinstance(r, dict)
-    assert len(r) == 186
+    assert len(r) == md_num
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 

--- a/tests/numpy_fs/test_numpy_fs_metadata.py
+++ b/tests/numpy_fs/test_numpy_fs_metadata.py
@@ -68,12 +68,15 @@ def test_numpy_fs_values_metadata_internal():
 def test_numpy_fs_metadata_keys():
     ds, _ = load_numpy_fs(1)
 
-    num = 183
+    # The number/order of metadata keys can vary with the ecCodes version.
+    # The same is true for the namespaces.
+
     md = ds[0].metadata()
-    assert len(md) == num
+    md_num = len(md)
+    assert md_num > 100
 
     keys = md.keys()
-    assert len(keys) == num
+    assert len(keys) == md_num
 
     for k in md.keys():
         assert isinstance(k, str)
@@ -81,7 +84,7 @@ def test_numpy_fs_metadata_keys():
         break
 
     items = md.items()
-    assert len(items) == num
+    assert len(items) == md_num
 
     for k, v in md.items():
         assert isinstance(k, str)
@@ -90,6 +93,7 @@ def test_numpy_fs_metadata_keys():
         break
 
     assert "max" not in md
+    assert "maximum" not in md
     assert "statistics.max" not in md
     assert "validityDate" in md
 
@@ -118,27 +122,31 @@ def test_numpy_fs_metadata_namespace():
     }
     assert r == ref
 
+    # The number/order of metadata keys can vary with the ecCodes version.
+    # The same is true for the namespaces.
+
     r = f[0].metadata(namespace=None)
     assert isinstance(r, dict)
-    assert len(r) == 177
+    md_num = len(r)
+    assert md_num > 100
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 
     r = f[0].metadata(namespace=[None])
     assert isinstance(r, dict)
-    assert len(r) == 177
+    assert len(r) == md_num
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 
     r = f[0].metadata(namespace="")
     assert isinstance(r, dict)
-    assert len(r) == 177
+    assert len(r) == md_num
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 
     r = f[0].metadata(namespace=[""])
     assert isinstance(r, dict)
-    assert len(r) == 177
+    assert len(r) == md_num
     assert r["level"] == 1000
     assert r["stepType"] == "instant"
 

